File: Data Structures & Algorithms/longest-common-prefix/submission-2.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        string referenceWord = strs[0];

        for (int i = 1; i < strs.Length; i++) {
            int j = 0;

            while (j < Math.Min(referenceWord.Length, strs[i].Length)) {
                if (referenceWord[j] != strs[i][j]) {
                    break;
                }

                j++;
            }

            referenceWord = referenceWord.Substring(0, j);
        }

        return referenceWord;
    }
}

File: Data Structures & Algorithms/merge-strings-alternately/submission-2.java
class Solution {
  public String mergeAlternately(String word1, String word2) {
    int min = Math.min(word1.length(), word2.length());
    char[] chars = new char[word1.length() + word2.length()];

    for (int i = 0; i < min; i++) {
        chars[2 * i] = word1.charAt(i);
        chars[2 * i + 1] = word2.charAt(i);
    }

    int index = 2 * min;
    String longer = word1.length() > word2.length() ? word1 : word2;

    for (int i = min; i < longer.length(); i++) {
        chars[index++] = longer.charAt(i);
    }

    return new String(chars);
}
}